### DISCUSSION TITLE
how to set a node's Label which is obtained by calling a Java method balck and bold?

### DISCUSSION THREAD

**Discussion by Willia-Yang** [eclipse-sirius/sirius-desktop#416] [Q&A]
The value of labelname is a string returned by calling a Java method. I want some of the fields to be bold and black. Is there any way to achieve this? For example, embed a markdown or html language in the string.
![图片](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/b88e0119-17ff-4787-bcad-b6918ab88a06)

**Comment by lredor** [comment]
Hi,
It is not possible to use specific "embedded markdown or HTML tag". But you can use conditionnal style:

- One default style without bold.
- A conditionnal style using a predicate expression (that can be another java method) with a bold style.
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/d6c818ac-0391-4898-b566-b87910bb57bb)

It is a little bit more explained in the [Sirius documentation](https://eclipse.dev/sirius/doc/specifier/diagrams/Diagrams.html#conditional_styles) .

Regards,

Laurent

**Reply by Willia-Yang** [reply]
I tried the way you recommended, according to my understanding, even if I add a conditional style, only one style will take effect, so it still not meet my requirements. Maybe you misunderstand me, All the contents in red line ”《part》part1S : PartDef1S [m] subsets part1” are a string as a whole, then i just want "subsets" to be bold.
![图片](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/42508a8d-56c4-4ee9-a3a9-948f093ddc42)

**Reply by lredor** [reply]
Indeed, I misunderstood. So, it is not natively feasible in Sirius. It might be possible by providing a custom EditPart and associated figure, but that would require development on your side. I am not aware of anyone who has done this before.

**Reply by Willia-Yang** [reply]
if i want to realize it  by providing a custom EditPart and associated figure, how should I start?  Any references or suggestions?

**Reply by lredor** [reply]
This [documentation](https://eclipse.dev/sirius/doc/developer/extensions-provide_custom_style.html) is the starting point. I don't know your knowledges about [GMF](https://eclipse.dev/modeling/gmp/?project=gmf-runtime) and [GEF](https://projects.eclipse.org/projects/tools.gef) frameworks, but without a minimal knowledge, it could be probably hard to be done.

**Reply by Willia-Yang** [reply]
ok, thanks
